sql/src/test/java/io/crate/planner/operators/JoinTest.java

```diff
@@ -109,6 +111,31 @@ public void testNestedLoop_TablesAreSwitchedIfLeftIsSmallerThanRight() {
         assertThat(((Reference) ((Collect) nl.left()).collectPhase().toCollect().get(0)).ident().tableIdent().name(), is("users"));
     }
 
+    @Test
+    public void testNestedLoop_TablesAreNotSwitchedIfLeftHasAPushedDownOrderBy() {
+        txnCtx.sessionContext().setHashJoinEnabled(false);
+        // we use a subselect to simulate the pushed-down order by
+        MultiSourceSelect mss = e.analyze("select users.id from (select id from users order by id) users, " +
+                                          "locations where users.id = locations.id");
+
+        TableStats tableStats = new TableStats();
+        ObjectObjectHashMap<RelationName, TableStats.Stats> rowCountByTable = new ObjectObjectHashMap<>();
+        rowCountByTable.put(TableDefinitions.USER_TABLE_IDENT, new TableStats.Stats(10, 0));
+        rowCountByTable.put(TableDefinitions.TEST_DOC_LOCATIONS_TABLE_IDENT, new TableStats.Stats(10_0000, 0));
+        tableStats.updateTableStats(rowCountByTable);
+
+        PlannerContext context = e.getPlannerContext(clusterService.state());
+        LogicalPlanner logicalPlanner = new LogicalPlanner(functions, tableStats);
+        SubqueryPlanner subqueryPlanner = new SubqueryPlanner((s) -> logicalPlanner.planSubSelect(s, context));
+        LogicalPlan operator = JoinPlanBuilder.createNodes(mss, mss.where(), subqueryPlanner, e.functions(), txnCtx)
+            .build(tableStats, Collections.emptySet());
+        Join nl = (Join) operator.build(
```

Comment:
not related to this PR, but it would be really awesome if we could extract all this boilerplate code above into a method as it's almost repeated at every test of this suite..